biogeochem/EDCanopyStructureMod.F90

```diff
@@ -499,7 +525,7 @@ subroutine DemoteFromLayer(currentSite,currentPatch,i_lyr)
                   currentSite%demotion_rate(currentCohort%size_class) = &
                        currentSite%demotion_rate(currentCohort%size_class) + currentCohort%n
                   currentSite%demotion_carbonflux = currentSite%demotion_carbonflux + &
-                       currentCohort%b_total() * currentCohort%n
+                       (leaf_c + store_c + fnrt_c + sapw_c + struct_c) * currentCohort%n
```

Comment:
yeah, I think it would be good.   The only reason I have not introduced this yet is because it would make the GetState function a little more complicated, and potentially slower because it would add more logic to it. Since it is used so much, and inside cohort loops, I wanted to keep it fast.   But I think there are ways to do this and not slow the code down...